internal/controller/stages/git.go

```diff
@@ -40,59 +41,70 @@ func (r *reconciler) getLatestCommits(
 			logger.Debug("found no credentials for git repo")
 		}
 
-		commit, err := r.getLatestCommitIDFn(sub.RepoURL, sub.Branch, repoCreds)
+		gm, err := r.getLatestCommitIDFn(ctx, sub.RepoURL, sub.Branch, repoCreds)
 		if err != nil {
 			return nil, errors.Wrapf(
 				err,
 				"error determining latest commit ID of git repo %q",
 				sub.RepoURL,
 			)
 		}
-		logger.WithField("commit", commit).
+		logger.WithField("commit", gm.Commit).
 			Debug("found latest commit from repo")
 		latestCommits[i] = kargoapi.GitCommit{
 			RepoURL: sub.RepoURL,
-			ID:      commit,
+			ID:      gm.Commit,
 			Branch:  sub.Branch,
+			Message: gm.Message,
 		}
 	}
 	return latestCommits, nil
 }
 
 func getLatestCommitID(
```

Comment:
Can we change the name of this function to reflect what it now returns? getLastestCommitMeta or similar?